File: 1480-running-sum-of-1d-array/1480-running-sum-of-1d-array.cpp
class Solution {
public:
    vector<int> runningSum(vector<int>& nums) {
        vector<int> r;
        r.push_back(nums[0]);
        for(int i=1; i<nums.size(); i++){
            r.push_back(nums[i]+r[i-1]);
        }
        return r;
    }
};

File: 1480-running-sum-of-1d-array/1480-running-sum-of-1d-array.cs
public class Solution {
    public int[] RunningSum(int[] nums) {
        for(int i = 1; i < nums.Length; i++)  nums[i] += nums[i-1];
        //int n = nums.Length-1;
        //nums[n] = nums[n] + sum(ref nums, n-1);
        return nums;
    }
    
   /* public int sum(ref int[] nums, int n) {
        if(n == 0)  return nums[0];
        nums[n] = nums[n] + sum(ref nums, n-1);
        return nums[n];
    }*/
}